codalab/lib/bundle_util.py

```diff
@@ -37,6 +37,44 @@ def bundle_to_bundle_info(model, bundle):
     return info
 
 
+def get_ancestors_string_representation(client, bundle_uuid):
+    """
+    Traverses up the dependency graph and compiles a nested list of the bundle's ancestors
+    :param client: Instance of JsonApiClient
+    :param bundle_uuid: UUID of bundle
+    :return: String representation of nested list
+    :rtype: String
+    """
+    bundle_cache = {}
+    seen = {bundle_uuid}
+    result = []
+
+    def get_bundle(uuid):
+        # check the cache before querying for a bundle
```

Comment:
This check is not necessary, since we are visiting each bundle just once. However, if we wanted to do a different representation of dependency graph, this check might be needed.